CRM/ACL/BAO/AclRole.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+
+/**
+ *  Access Control AclRole.
+ */
+class CRM_ACL_BAO_AclRole extends CRM_ACL_DAO_AclRole {
+  public static $_entityTable = NULL;
+
+  /**
+   * Whitelist of possible values for the entity_table field
+   *
+   * @return array
+   */
+  public static function entityTables(): array {
+    if (!self::$_entityTable) {
+      self::$_entityTable = [
+        'civicrm_contact' => ts('Contact'),
+        'civicrm_group' => ts('Group'),
+      ];
+    }
+    return self::$_entityTable;
+  }
+
+  /**
+   * @param array $params
+   *
+   * @return CRM_ACL_DAO_EntityRole
+   */
+  public static function create(&$params) {
+    $dao = new CRM_ACL_DAO_EntityRole();
```

Comment:
should we not use the writeRecord function here?

Reply:
@seamuslee001 I think this is just the existing file renamed